library/pod-security-policy/host-namespaces/template.yaml

```diff
@@ -26,29 +26,41 @@ spec:
             https://kubernetes.io/docs/concepts/policy/pod-security-policy/#host-namespaces
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8spsphostnamespace
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: sharingNamespace
+            expression: |
+              has(variables.anyObject.spec.hostPID) || has(variables.anyObject.spec.hostIPC) ? variables.anyObject.spec.hostPID || variables.anyObject.spec.hostIPC : false
```

Comment:
I think this is vulnerable to failing on nil.

Suggest splitting this in to two variables:

* get hostIPC if exists, false otherwise
* same for hostPID
* test the downstream variables

Reply:
Updated the code. PTAL @maxsmythe